frontend/readline.c

```diff
@@ -37,10 +43,39 @@ static void sigint_handler(int n) {
 }
 #endif
 
+
+static void mkdirs(const char *path) {
```

Comment:
Maybe `mkdir_all` would be a better name? Would make it easier to tell it apart from `mkdir`.